### PR TITLE
feat: Check assets in bridges before claim in mainnet

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: src/pages/api/claim.ts [hunk 1]
```diff
@@ -57,14 +58,22 @@ export default async function handler(
...
   const from = signer.address;
 
-  logger.info(`[claim] fromAddress: ${from}, toAddress: ${account}, amount(pCKB): ${claimValueInEth}`);
+  if (env.IS_MAINNET) {
+    // check assets in bridges
+    const flag = await hasEnoughUSD(account);
+    if (!flag) {
+      logger.info(`account ${account} don't have enough USD`)
+      return res.status(200).json({
+        message: `Don't have enough USD in bridges`,
```
Comment (Flouse) [inline src/pages/api/claim.ts]:
```suggestion
        message: `To stop spam attacks draining the faucet, the account must contain assets such as USDC or USDT that have been bridged from Celer cBridge or Multichain.`
```

File: src/pages/api/claim.ts [hunk 2]
```diff
@@ -57,14 +58,22 @@ export default async function handler(
   const signer = new ethers.Wallet(privateKey, provider);
   const from = signer.address;
 
-  logger.info(`[claim] fromAddress: ${from}, toAddress: ${account}, amount(pCKB): ${claimValueInEth}`);
+  if (env.IS_MAINNET) {
+    // check assets in bridges
+    const flag = await hasEnoughUSD(account);
+    if (!flag) {
+      logger.info(`account ${account} don't have enough USD`)
+      return res.status(200).json({
```
Comment (Flouse) [inline src/pages/api/claim.ts]: Maybe we should use a `40x` code?

File: src/lib/bridges/prices/average.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  try {
+    cryptocomPriceInfos = await cryptocomPrices(allSymbols);
+  } catch (err) {
+    logger.error(`request to crypto.com failed`)
+    failedPriceInfos += 1;
+  }
+
+  if (failedPriceInfos === 0) {
```
Comment (Flouse) [inline src/lib/bridges/prices/average.ts]:
```suggestion
  if (failedPriceInfos === 3) {
```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: src/pages/api/claim.ts (hunk 2 above)
Comment (classicalliu) [reply src/pages/api/claim.ts]: Updated to `422`

**Review by Flouse — APPROVED** [review]

**Pull request merged**